modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C

```diff
@@ -165,25 +218,49 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::LAROMANCEStressUpdateBaseTempl(
 {
-  this->_check_range = true;
-
-  _window_failure[_cell_input_index] =
-      parameters.get<MooseEnum>("cell_input_window_failure").getEnum<WindowFailure>();
-  _window_failure[_wall_input_index] =
-      parameters.get<MooseEnum>("wall_input_window_failure").getEnum<WindowFailure>();
-  _window_failure[_stress_input_index] =
-      parameters.get<MooseEnum>("stress_input_window_failure").getEnum<WindowFailure>();
-  _window_failure[_old_strain_input_index] =
-      parameters.get<MooseEnum>("old_strain_input_window_failure").getEnum<WindowFailure>();
-  _window_failure[_temperature_input_index] =
-      parameters.get<MooseEnum>("temperature_input_window_failure").getEnum<WindowFailure>();
+  this->_check_range = true; // todo, this may not be necessary?
+
+  for (unsigned int i = 0; i < _window_failure.size(); ++i)
+    _window_failure[i].resize(2);
+
+  _window_failure[_cell_input_index][0] =
+      parameters.get<MooseEnum>("cell_input_window_low_failure").getEnum<WindowFailure>();
+  _window_failure[_cell_input_index][1] =
+      parameters.get<MooseEnum>("cell_input_window_high_failure").getEnum<WindowFailure>();
+  _window_failure[_wall_input_index][0] =
+      parameters.get<MooseEnum>("wall_input_window_low_failure").getEnum<WindowFailure>();
+  _window_failure[_wall_input_index][1] =
+      parameters.get<MooseEnum>("wall_input_window_high_failure").getEnum<WindowFailure>();
+  _window_failure[_stress_input_index][0] =
+      parameters.get<MooseEnum>("stress_input_window_low_failure").getEnum<WindowFailure>();
+  _window_failure[_stress_input_index][1] =
+      parameters.get<MooseEnum>("stress_input_window_high_failure").getEnum<WindowFailure>();
+  _window_failure[_old_strain_input_index][0] =
+      parameters.get<MooseEnum>("old_strain_input_window_low_failure").getEnum<WindowFailure>();
+  _window_failure[_old_strain_input_index][1] =
+      parameters.get<MooseEnum>("old_strain_input_window_high_failure").getEnum<WindowFailure>();
+  _window_failure[_temperature_input_index][0] =
+      parameters.get<MooseEnum>("temperature_input_window_low_failure").getEnum<WindowFailure>();
+  _window_failure[_temperature_input_index][1] =
+      parameters.get<MooseEnum>("temperature_input_window_high_failure").getEnum<WindowFailure>();
   if (_environmental)
-    _window_failure[_environmental_input_index] =
-        parameters.get<MooseEnum>("environment_input_window_failure").getEnum<WindowFailure>();
+  {
+    _window_failure[_environmental_input_index][0] =
+        parameters.get<MooseEnum>("environment_input_window_low_failure").getEnum<WindowFailure>();
+    _window_failure[_environmental_input_index][1] =
+        parameters.get<MooseEnum>("environment_input_window_high_failure").getEnum<WindowFailure>();
+  }
 
   setupUnitConversionFactors(parameters);
 }
 
+template <bool is_ad>
+bool
+LAROMANCEStressUpdateBaseTempl<is_ad>::substeppingCapabilityEnabled()
+{
+  return this->template getParam<bool>("use_substep");
```

Comment:
FYI @dschwen this is required (return true led to errors?)

Reply:
Got it. That also means the function name is good. Not how I would have designed it though, as the parameter is added by a parent class and I prefer parameter handling to happen at the same level.